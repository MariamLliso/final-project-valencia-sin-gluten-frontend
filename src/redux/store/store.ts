import { configureStore } from "@reduxjs/toolkit";
import establishmentsSliceReducer from "../features/establishmentsSlice/establishmentsSlice";
import singleEstablishmentSlice from "../features/singleEstablishmentSlice/singleEstablishmentSlice";
import uiSliceReducer from "../features/uiSlice/uiSlice";
import userProfileSliceReducer from "../features/userProfileSlice/userProfileSlice";
import userSliceReducer from "../features/userSlice/userSlice";

const store = configureStore({
  reducer: {
    ui: uiSliceReducer,
    user: userSliceReducer,
    userProfile: userProfileSliceReducer,
    establishments: establishmentsSliceReducer,
    singleEstablishment: singleEstablishmentSlice,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
