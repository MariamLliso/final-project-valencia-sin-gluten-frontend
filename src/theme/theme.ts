import { alpha, createTheme } from "@mui/material";

export const pxToRem = (value: number) => {
  return `${value / 16}rem`;
};

interface IMediaSizes {
  sm: number;
  md: number;
  lg: number;
}

export const responsiveFontSizes = ({ sm, md, lg }: IMediaSizes) => {
  return {
    "@media (min-width:600px)": {
      fontSize: pxToRem(sm),
    },
    "@media (min-width:900px)": {
      fontSize: pxToRem(md),
    },
    "@media (min-width:1200px)": {
      fontSize: pxToRem(lg),
    },
  };
};

export const fontPrimary = "Poppins, sans-serif";
export const fontSecondary = "Libre Bodoni, serif;";

const greyColor = {
  0: "#FFFFFF",
  100: "#F5F5F5",
  200: "#F4F6F8",
  300: "#DFE3E8",
  400: "#C4CDD5",
  500: "#919EAB",
  600: "#637381",
  700: "#454F5B",
  800: "#212B36",
  900: "#161C24",
  500_8: alpha("#919EAB", 0.08),
  500_12: alpha("#919EAB", 0.12),
  500_16: alpha("#919EAB", 0.16),
  500_24: alpha("#919EAB", 0.24),
  500_32: alpha("#919EAB", 0.32),
  500_48: alpha("#919EAB", 0.48),
  500_56: alpha("#919EAB", 0.56),
  500_80: alpha("#919EAB", 0.8),
};

const primaryColor = {
  light: "#E9F3D9",
  main: "#5FAA46",
  dark: "#3b8224",
  contrastText: "#fff",
};

const secondaryColor = {
  lighter: "#D6E4FF",
  light: "#84A9FF",
  main: "#3366FF",
  dark: "#1939B7",
  darker: "#091A7A",
  contrastText: "#fff",
};

const theme = createTheme({
  typography: {
    fontFamily: fontPrimary,
    fontWeightRegular: 400,
    fontWeightMedium: 500,
    fontWeightBold: 600,
    h1: {
      fontFamily: fontSecondary,
      fontWeight: 700,
      lineHeight: 80 / 64,
      fontSize: pxToRem(40),
      ...responsiveFontSizes({ sm: 52, md: 58, lg: 64 }),
    },
    h2: {
      fontFamily: fontSecondary,
      fontWeight: 700,
      lineHeight: 64 / 48,
      fontSize: pxToRem(32),
      ...responsiveFontSizes({ sm: 40, md: 44, lg: 48 }),
    },
    h3: {
      fontFamily: fontSecondary,
      fontWeight: 700,
      lineHeight: 1.5,
      fontSize: pxToRem(24),
      ...responsiveFontSizes({ sm: 26, md: 30, lg: 32 }),
    },
    h4: {
      fontWeight: 700,
      lineHeight: 1.5,
      fontSize: pxToRem(20),
      ...responsiveFontSizes({ sm: 20, md: 24, lg: 24 }),
    },
    h5: {
      fontWeight: 700,
      lineHeight: 1.5,
      fontSize: pxToRem(18),
      ...responsiveFontSizes({ sm: 19, md: 20, lg: 20 }),
    },
    h6: {
      fontWeight: 700,
      lineHeight: 28 / 18,
      fontSize: pxToRem(17),
      ...responsiveFontSizes({ sm: 18, md: 18, lg: 18 }),
    },
    subtitle1: {
      fontWeight: 600,
      lineHeight: 1.5,
      fontSize: pxToRem(16),
    },
    subtitle2: {
      fontWeight: 600,
      lineHeight: 22 / 14,
      fontSize: pxToRem(14),
    },
    body1: {
      lineHeight: 1.5,
      fontSize: pxToRem(16),
    },
    body2: {
      lineHeight: 22 / 14,
      fontSize: pxToRem(14),
    },
    caption: {
      lineHeight: 1.5,
      fontSize: pxToRem(12),
    },
    overline: {
      fontWeight: 700,
      lineHeight: 1.5,
      fontSize: pxToRem(12),
      letterSpacing: 1.1,
      textTransform: "uppercase",
    },
    button: {
      fontWeight: 700,
      lineHeight: 24 / 14,
      fontSize: pxToRem(14),
      textTransform: "uppercase",
    },
  },
  palette: {
    common: { black: "#000", white: "#fff" },
    primary: { ...primaryColor },
    secondary: { ...secondaryColor },
    grey: greyColor,
    text: {
      primary: greyColor[800],
      secondary: greyColor[600],
      disabled: greyColor[500],
    },
    background: { paper: "#fff", default: greyColor[0] },
    action: {
      active: greyColor[600],
      hover: greyColor[500_8],
      selected: greyColor[500_16],
      disabled: greyColor[500_80],
      disabledBackground: greyColor[500_24],
      focus: greyColor[500_24],
      hoverOpacity: 0.08,
      disabledOpacity: 0.48,
    },
  },
  components: {
    MuiButtonBase: {
      styleOverrides: {
        root: {
          letterSpacing: "0.1em",
        },
      },
      defaultProps: {
        disableRipple: true,
      },
    },
    MuiInputBase: {
      styleOverrides: {
        input: {
          "&::placeholder": {
            fontSize: "0.875rem",
          },
        },
      },
    },
    MuiOutlinedInput: {
      styleOverrides: {
        root: {
          borderRadius: `10px`,
          "&.MuiInputBase-multiline": {
            padding: 1,
          },
        },
        input: {
          fontWeight: 500,
          padding: "15.5px 14px",
          borderRadius: `10px`,
          "&.MuiInputBase-inputSizeSmall": {
            padding: "10px 14px",
            "&.MuiInputBase-inputAdornedStart": {
              paddingLeft: 0,
            },
          },
        },
        inputAdornedStart: {
          paddingLeft: 4,
        },
        notchedOutline: {
          borderRadius: `10px`,
        },
      },
    },
  },
});

export default theme;
