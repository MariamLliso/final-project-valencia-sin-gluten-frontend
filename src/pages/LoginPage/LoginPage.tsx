import { Typography } from "@mui/material";
import { useEffect } from "react";
import { useNavigate } from "react-router-dom";
import LoginForm from "../../components/LoginForm/LoginForm";
import { useAppSelector } from "../../redux/hooks";
import { RootState } from "../../redux/store/store";
import { IUserState } from "../../types/userInterfaces";
import LoginPageStyle from "./LoginPageStyle";

const LoginPage = (): JSX.Element => {
  const { logged }: IUserState = useAppSelector(
    (state: RootState) => state.user
  );
  const navigate = useNavigate();

  useEffect(() => {
    if (logged) {
      navigate("/user/profile");
    }
  }, [navigate, logged]);

  return (
    <LoginPageStyle>
      <Typography variant="h1" component="h2" className="register_tittle">
        ¡Nos alegra verte de nuevo por aquí!
      </Typography>
      <LoginForm></LoginForm>
    </LoginPageStyle>
  );
};

export default LoginPage;
