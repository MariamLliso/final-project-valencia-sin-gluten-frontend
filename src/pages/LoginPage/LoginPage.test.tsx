import { render, screen, waitFor } from "@testing-library/react";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";
import store from "../../redux/store/store";
import theme from "../../theme/theme";
import TestRenderer from "react-test-renderer";
import LoginPage from "./LoginPage";
import { BrowserRouter } from "react-router-dom";

const mockUseNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockUseNavigate,
}));

describe("Given the loginpage component", () => {
  describe("When it's invoked", () => {
    test("Then it should always match this snapshot", () => {
      const testedRegisterPage = TestRenderer.create(
        <BrowserRouter>
          <Provider store={store}>
            <ThemeProvider theme={theme}>
              <LoginPage />
            </ThemeProvider>
          </Provider>
        </BrowserRouter>
      );

      expect(testedRegisterPage).toMatchSnapshot();
    });

    test("Then it should render a register form with a button with the text 'iniciar sesión'", () => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <ThemeProvider theme={theme}>
              <LoginPage />
            </ThemeProvider>
          </Provider>
        </BrowserRouter>
      );

      const expectedButton = screen.getByRole("button", {
        name: "iniciar sesión",
      });

      expect(expectedButton).toBeInTheDocument();
    });
  });

  describe("When invoked but user is already logged", () => {
    test("Then it should navigate to '/user/profile'", async () => {
      const mockPayload = {
        type: "user/login",
        payload: {
          id: "62a260eeb80d5435426c9163",
          username: "mariamadmin",
          userRol: "ADM",
          iat: 1655307037,
        },
      };
      await waitFor(() => {
        const loginAction = {
          type: "user/login",
          payload: mockPayload,
        };

        store.dispatch(loginAction);
      });

      render(
        <BrowserRouter>
          <Provider store={store}>
            <ThemeProvider theme={theme}>
              <LoginPage />
            </ThemeProvider>
          </Provider>
        </BrowserRouter>
      );

      expect(mockUseNavigate).toHaveBeenLastCalledWith("/user/profile");
    });
  });
});
