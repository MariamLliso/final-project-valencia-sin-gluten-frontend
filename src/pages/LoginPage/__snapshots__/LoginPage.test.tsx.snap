// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the loginpage component When it's invoked Then it should always match this snapshot 1`] = `
<div
  className="sc-gsnTZi esGYSl"
>
  <h2
    className="MuiTypography-root MuiTypography-h1 register_tittle css-o2w69a-MuiTypography-root"
  >
    ¡Nos alegra verte de nuevo por aquí!
  </h2>
  <form
    autoComplete="off"
    onSubmit={[Function]}
  >
    <div
      className="css-1lwbda4-MuiStack-root"
    >
      <div
        className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="username"
        >
          Username
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="username"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1ftyaf0"
            >
              <span>
                Username
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          id="username-helpertext"
        >
          <span
            className="notranslate"
          >
            ​
          </span>
        </p>
      </div>
      <div
        className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="password"
        >
          Password
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
          role="password"
        >
          <input
            aria-invalid={false}
            className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="password"
            value=""
          />
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <button
              aria-label="toggle-password-visibility"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="VisibilityIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                />
              </svg>
            </button>
          </div>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1ftyaf0"
            >
              <span>
                Password
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
          id="password-helpertext"
        >
          <span
            className="notranslate"
          >
            ​
          </span>
        </p>
      </div>
      <button
        className="MuiLoadingButton-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButtonBase-root css-sppx0e-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
        disabled={false}
        id=":r0:"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        iniciar sesión
      </button>
      <button
        className="MuiTypography-root MuiTypography-subtitle2 MuiLink-root MuiLink-underlineNone MuiLink-button css-1kyronh-MuiTypography-root-MuiLink-root"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="button"
      >
        ¿Aun no tienes cuenta? ¡Registrate ahora!
      </button>
    </div>
  </form>
</div>
`;
